tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -1861,9 +1870,7 @@ def feature_importance(self, data=None, model=None, features=None, feature_stage
         if (data is None) and (not self._trainer.is_data_saved):
             raise AssertionError(
                 'No data was provided and there is no cached data to load for feature importance calculation. `cache_data=True` must be set in the `TabularPredictor` init `learner_kwargs` argument call to enable this functionality when data is not specified.')
-        if data is not None:
-            # Avoid crash when indices are duplicated
-            data = data.reset_index(drop=True)
+        self._validate_unique_indices(data, 'data')
```

Comment:
I forgot I had already added a work-around to fix this via the .reset_index logic.

This is a bit tricky in terms of which we want:

1. Encourage best practice from users and error if they have invalid data
2. Make it work even if they have invalid data
3. Same as 2 but log a warning.

Curious on thoughts.

Reply:
I think raise exception is better if the user simply provides invalid data. Making it to work in our way might not be what the user truly wants. Making it work with a warning is reasonable, but my concern is user would simply neglect the warning because of the amount of logs autogluon give and don't know what we've done.

Reply:
I think that makes sense, let's raise an exception.